### PR TITLE
#ARRUS-197: Exposed the possibility to configure and program multiple probes and sequences. 

### PR DESCRIPTION
**Pull request by pjarosik**
NOTE, the following work should be done before releasing 0.12.0:
- improve the unit test coverage,
- perform the system tests using vandv.

The above issues will be solved during the course of implementing v0.12.0. Still, it's ready to be merged and tested on the branch v0.12.0-dev.

### CODE REVIEW

File: api/python/examples/classical_beamforming.py [hunk 1]
```diff
@@ -15,37 +15,31 @@
 arrus.add_log_file("test.log", arrus.logging.INFO)
 
 # Here starts communication with the device.
-with arrus.Session() as sess:
+with arrus.Session("/home/pjarosik/us4r.prototxt") as sess:
```
Comment (jrozb91) [inline api/python/examples/classical_beamforming.py:18]: Path to prototxt has to be removed

File: api/python/examples/custom_tx_rx_sequence.py [hunk 2]
```diff
@@ -40,48 +40,40 @@
 def main():
     # Here starts communication with the device.
     medium = arrus.medium.Medium(name="water", speed_of_sound=1490)
-    with arrus.Session("us4r.prototxt", medium=medium) as sess:
+    with arrus.Session("/home/pjarosik/us4r.prototxt", medium=medium) as sess:
```
Comment (jrozb91) [inline api/python/examples/custom_tx_rx_sequence.py:43]: Path to prototxt has to be removed

File: api/python/examples/old_test.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+)
+
+arrus.set_clog_level(arrus.logging.INFO)
+arrus.add_log_file("test.log", arrus.logging.INFO)
+
+
+def main():
+    # Here starts communication with the device.
+    medium = arrus.medium.Medium(name="water", speed_of_sound=1490)
+    with arrus.Session("/home/pjarosik/us4r.prototxt", medium=medium) as sess:
```
Comment (jrozb91) [inline api/python/examples/old_test.py:43]: Path to prototxt has to be removed

File: api/python/CMakeLists.txt [hunk 4]
```diff
@@ -226,17 +226,17 @@ install(
...
+#    create_python_venv(py_tests_venv ${CMAKE_CURRENT_BINARY_DIR})
+#    install_arrus_package(py_tests_venv_arius py_tests_venv python_whl "DIR")
+#    install_cupy_package(py_tests_venv_cupy py_tests_venv ${CUDAToolkit_VERSION})
+#    get_target_property(PY_TESTS_VENV_EXEC py_tests_venv VENV_EXECUTABLE)
+#    foreach(TEST_FILE IN LISTS TEST_FILES)
+#        add_test(
+#            NAME ${TEST_FILE}
+#            COMMAND ${PY_TESTS_VENV_EXEC} -m unittest ${CMAKE_CURRENT_BINARY_DIR}/${TEST_FILE})
+#    endforeach()
+#endif()
```
Comment (jrozb91) [inline api/python/CMakeLists.txt:242]: Why commented out?

File: api/python/examples/custom_tx_rx_sequence.py (hunk 2 above)
Comment (pjarosik) [reply api/python/examples/custom_tx_rx_sequence.py]: Fixed

File: api/python/examples/classical_beamforming.py (hunk 1 above)
Comment (pjarosik) [reply api/python/examples/classical_beamforming.py]: Fixed

File: api/python/examples/old_test.py (hunk 3 above)
Comment (pjarosik) [reply api/python/examples/old_test.py:43]: Fixed

File: api/python/examples/diverging_beams.py [hunk 5]
```diff
@@ -14,7 +14,7 @@
 arrus.add_log_file("test.log", arrus.logging.INFO)
 
 # Here starts communication with the device.
-with arrus.Session() as sess:
+with arrus.Session("/home/pjarosik/us4r.prototxt") as sess:
```
[the remark this answers is not in the document]
Comment (pjarosik) [reply api/python/examples/diverging_beams.py]: Fixed

File: api/python/CMakeLists.txt (hunk 4 above)
Comment (pjarosik) [reply api/python/CMakeLists.txt:242]: Good point! to enable this Python unit tests requires update, I will do it ASAP

**Comment by pjarosik** [comment]
@zkLog moving back to in progress, as there are some changes required here

**Comment by pjarosik** [comment]
replaced by #323

**Pull request closed without merging**